app/src/main/java/org/openedx/app/MainFragment.kt

```diff
@@ -81,11 +83,20 @@ class MainFragment : Fragment(R.layout.fragment_main) {
             }
         }
 
-        requireArguments().apply {
-            this.getString(ARG_COURSE_ID, null)?.apply {
-                router.navigateToCourseDetail(parentFragmentManager, this)
+        requireArguments().run {
+            getString(ARG_COURSE_ID)?.let { courseId ->
```

Comment:
We can implement something along the lines of :
`getString(ARG_COURSE_ID).takeIf { it.isNullOrBlank().not() }?.let { courseId ->`

Otherwise, the user will get stuck in a loop because we are using an empty string to clear, which will always be non-null.